### PR TITLE
Migrate to esConsumes some more vertexing-related packages

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

Part of #31061.
Migrated to esConsumes the following pakages:
  * `RecoPixelVertexing/PixelVertexFinding`
  * `RecoVertex/V0Producer`
  * `RecoVertex/KalmanVertexFit`

Profited to transform also some legacy `EDAnalyzer`s to `one::EDAnalyzer`. 

#### PR validation:

It builds

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Not a backport.

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

File: RecoVertex/V0Producer/src/V0Fitter.h [hunk 1]
```diff
@@ -46,6 +46,8 @@ class dso_hidden V0Fitter {
               reco::VertexCompositeCandidateCollection& l);
 
 private:
+  const edm::ESGetToken<MagneticField, IdealMagneticFieldRecord> esToken_MF;
```
Comment (slava77) [inline RecoVertex/V0Producer/src/V0Fitter.h:49]: almost all data members use the (recommended) trailing underscore in the name. Please consider to update here as well.

Comment (mmusich) [reply RecoVertex/V0Producer/src/V0Fitter.h]: done in the last push.

**Comment by mmusich** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+reconstruction

for https://github.com/cms-sw/cmssw/pull/33232	c4551bf
- code changes are in line with the PR description and the follow up review; the update is technical (no changes to the physics performance)
- jenkins tests pass and comparisons with the baseline show no (relevant) differences

**Comment by qliphy** [comment]
+1

**Pull request merged**